### PR TITLE
Added one diode model

### PR DESCRIPTION
**Pull request by antonszilasi**

### CODE REVIEW

File: src/Honeybee_Generator_PV.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # 
 # This file is part of Honeybee.
 # 
-# Copyright (c) 2013-2016, Anton Szilasi <ajszilas@gmail.com>
```
Comment (mostaphaRoudsari) [inline src/Honeybee_Generator_PV.py]: It should be 2016

File: src/Honeybee_Generator_PV.py [hunk 2]
```diff
@@ -22,21 +22,17 @@
 
 
 """
-Provided by Honeybee 0.0.59
-
+Provided by Honeybee 0.0.58
```
Comment (mostaphaRoudsari) [inline src/Honeybee_Generator_PV.py]: @antonszilasi the version is wrong!

File: src/Honeybee_Generator_PV.py [hunk 3]
```diff
@@ -22,21 +22,17 @@
...
 For each Photovoltaic generator there must be a inverter for power to be produced, if several photovoltaic generators are modelled in the same generatorsystem (Connected to the generationsystem component) these generators must have the same inverter.
-
 For more information about Photovolatic generators please see: 
 -
 http://bigladdersoftware.com/epx/docs/8-2/input-output-reference/group-electric-load-center.html#photovoltaic-generators
-
 -
-Provided by Honeybee 0.0.59
-
+Provided by Honeybee 0.0.58
```
Comment (mostaphaRoudsari) [inline src/Honeybee_Generator_PV.py]: The same here.

File: src/Honeybee_Generator_PV.py [hunk 4]
```diff
@@ -57,11 +53,10 @@
 
 ghenv.Component.Name = "Honeybee_Generator_PV"
 ghenv.Component.NickName = 'PVgen'
-ghenv.Component.Message = 'VER 0.0.59\nJAN_26_2016'
-ghenv.Component.IconDisplayMode = ghenv.Component.IconDisplayMode.application
+ghenv.Component.Message = 'VER 0.0.58\nFEB_01_2016'
 ghenv.Component.Category = "Honeybee"
-ghenv.Component.SubCategory = "13 | WIP" #"06 | Honeybee"
-#compatibleHBVersion = VER 0.0.56\nJAN_16_2016
+ghenv.Component.SubCategory = "12 | WIP" #"06 | Honeybee"
```
Comment (mostaphaRoudsari) [inline src/Honeybee_Generator_PV.py]: WIP is under 13 and not 12 anymore

File: src/Honeybee_Honeybee.py [hunk 5]
```diff
@@ -240,10 +240,8 @@ def checkForUpdates(self, LB= True, HB= True, OpenStudio = True, template = True
                 isNewerThermAvailable = True
             else:
                 # find the version
-                try:
-                    with open(thermFile) as tempFile:
-                        currentThermVersion = eval(tempFile.readline().split("!")[-1].strip())["version"]
-                except: isNewerThermAvailable = True
+                with open(thermFile) as tempFile:
```
Comment (mostaphaRoudsari) [inline src/Honeybee_Honeybee.py]: @antonszilasi this should be kept!

**Comment by antonszilasi** [comment]
@mostaphaRoudsari  Sorry for the previous issues this should do the trick.

File: src/Honeybee_Generator_PV.py [hunk 6]
```diff
@@ -57,11 +53,10 @@
 
 ghenv.Component.Name = "Honeybee_Generator_PV"
 ghenv.Component.NickName = 'PVgen'
-ghenv.Component.Message = 'VER 0.0.59\nJAN_26_2016'
-ghenv.Component.IconDisplayMode = ghenv.Component.IconDisplayMode.application
```
Comment (mostaphaRoudsari) [inline src/Honeybee_Generator_PV.py]: @antonszilasi sorry for being picky! but you're missing the new line for setting icon default to application.

File: src/Honeybee_Honeybee.py [hunk 7]
```diff
@@ -244,6 +244,9 @@ def checkForUpdates(self, LB= True, HB= True, OpenStudio = True, template = True
                     with open(thermFile) as tempFile:
                         currentThermVersion = eval(tempFile.readline().split("!")[-1].strip())["version"]
                 except: isNewerThermAvailable = True
+
+                with open(thermFile) as tempFile:
```
Comment (mostaphaRoudsari) [inline src/Honeybee_Honeybee.py]: This line looks duplicated! we either need this or 244 to 246. I believe the one on the top is the one to be kept.
Am I right @chriswmackey?

File: src/Honeybee_Generator_Wind_Horizontialaxis.py [hunk 8]
```diff
@@ -134,206 +134,212 @@ def checktheinputs(_name,_simpleOrAnalytical,_powerCoefficients,_powerControl,_r
         ghenv.Component.AddRuntimeMessage(w, "Only one template can be used at a time! Please set either TemplateLargeTurbine_ or TemplateMediumTurbine_ to true not both!")
         return -1
 
-    if _powerControl == None:
-        print "The field _powerControl must be specified!"
+    if powerControl_ == None:
```
Comment (mostaphaRoudsari) [inline src/Honeybee_Generator_Wind_Horizontialaxis.py]: if an input is required it should be _input and empty on the right side. The older version was correct. I'm not sure why are you changing them to input_? Is powerControl an optional input?

File: src/Honeybee_Generator_Wind_Horizontialaxis.py [hunk 9]
```diff
@@ -134,206 +134,212 @@ def checktheinputs(_name,_simpleOrAnalytical,_powerCoefficients,_powerControl,_r
...
+    if rotorSpeed_ == None:
+        print "The field rotorSpeed_ must be specified!"
         w = gh.GH_RuntimeMessageLevel.Warning
-        ghenv.Component.AddRuntimeMessage(w, "The field _rotorSpeed must be specified!")
+        ghenv.Component.AddRuntimeMessage(w, "The field rotorSpeed_ must be specified!")
         return -1
         
-    if _rotorDiameter == None:
-        print "The field _rotorDiameter must be specified!"
+    if rotorDiameter_ == None:
```
Comment (mostaphaRoudsari) [inline src/Honeybee_Generator_Wind_Horizontialaxis.py]: @antonszilasi the same here. If you need a reference you can always check this document - check number 4: https://github.com/mostaphaRoudsari/ladybug/blob/master/resources/Ladybug%20Grahpic%20Standards.md

File: src/Honeybee_Generator_Wind_Horizontialaxis.py [hunk 10]
```diff
@@ -134,206 +134,212 @@ def checktheinputs(_name,_simpleOrAnalytical,_powerCoefficients,_powerControl,_r
...
+    if localAvWindspeed_ == None:
+        print "The field localAvWindspeed_ must be specified!"
         w = gh.GH_RuntimeMessageLevel.Warning
-        ghenv.Component.AddRuntimeMessage(w, "The field _localAvWindspeed must be specified!")
+        ghenv.Component.AddRuntimeMessage(w, "The field localAvWindspeed_ must be specified!")
         return -1
 
-    if _turbinecost == None:
-        print "The field _turbinecost must be specified!"
+    if turbinecost_ == None:
```
Comment (mostaphaRoudsari) [inline src/Honeybee_Generator_Wind_Horizontialaxis.py]: I'm not sure why you change all this input names. Also you can use a honeybee method to check the inputs for you if you name them correctly. Here is an example: https://github.com/mostaphaRoudsari/Honeybee/blob/master/src/Honeybee_Radiance%20Glass%20Material.py#L90-L91

File: src/Honeybee_Generator_Wind_Horizontialaxis.py (hunk 8 above)
Comment (antonszilasi) [inline src/Honeybee_Generator_Wind_Horizontialaxis.py]: @mostaphaRoudsari 
Sorry the delay I lost my Grasshopper for about 10 days (was getting withdrawal symptoms). They are all optional inputs hence I had to change them. Are you able to merge this pull request now?

**Pull request closed without merging**